providers/npu/bin/check_firmware_version.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+import re
+import subprocess
+from pathlib import (
+    Path,
+)
+from typing import (
+    Optional,
+)
+
+FIRMWARE_SEARCH_DIR = Path("/var/snap/intel-npu-driver/current/intel/vpu")
+VERSION_PATTERN = re.compile(r"^(\d{8}\*|[A-Z][a-z]{2}\s+\d{1,2}\s+\d{4}\*).*")
+
+
+def get_active_firmware_line() -> Optional[str]:
+    result = subprocess.run(
+        [
+            "journalctl",
+            "--dmesg",
+        ],
+        capture_output=True,
+        text=True,
+        check=True,
+        encoding="utf-8",
+    )
+    all_lines = result.stdout.splitlines()
+
+    matching_lines = [
+        line for line in all_lines if "Firmware: intel/vpu" in line
+    ]
+
+    if not matching_lines:
+        raise SystemExit("No 'intel_vpu' firmware logs found in dmesg.")
+        return None
+
+    return matching_lines[-1]
+
+
+def find_version_in_file(
+    filepath: Path,
+) -> Optional[str]:
+    try:
+        result = subprocess.run(
+            [
+                "strings",
+                filepath,
+            ],
+            capture_output=True,
+            text=True,
+            check=True,
+            encoding="utf-8",
+        )
+        for line in result.stdout.splitlines():
+            # Return the first match found
+            if VERSION_PATTERN.match(line):
+                return line
+    except (
+        subprocess.CalledProcessError,
+        FileNotFoundError,
+    ):
+        # This can happen with corrupted files or if 'strings' isn't installed
+        return None
+    return None
```

Comment:
No need to retunr None in python.
I would log here the error though, to know that it could not read any file
```suggestion
        print("Version pattern not found in file")
```